specification/network/resource-manager/Microsoft.Network/stable/2019-06-01/webapplicationfirewall.json

```diff
@@ -686,6 +694,80 @@
           "description": "When matchVariable is a collection, operator used to specify which elements in the collection this exclusion applies to."
         }
       }
+    },
+    "BotDetection": {
+      "description": "Allows the detection of bots and specify action for each bot category.",
+      "required": [
+        "good", 
+        "neutral", 
+        "bad"
+      ],
+      "properties": {
+        "good": {
+          "description": "Describes the action to be taken when detecting a good bot.",
+          "properties": {
+            "action": {
+              "description": "Describes which action should be triggered by the bot category.",
+              "$ref": "#/definitions/Action"
+            }
+          }
+        },
+        "neutral": {
+          "description": "Describes the action to be taken when detecting a neutral bot.",
+          "properties": {
+            "action": {
+              "description": "Describes which action should be triggered by the bot category.",
+              "$ref": "#/definitions/Action"
+            }
+          }
+        },
+        "bad": {
+          "description": "Describes the action to be taken when detecting a bad bot.",
+          "properties": {
+            "action": {
+              "description": "Describes which action should be triggered by the bot category.",
+              "$ref": "#/definitions/Action"
+            }
+          }
+        },
+        "overrides": {
+          "description": "List of custom bot categories which take precedence over the predefined categories.",
+          "type": "array",
+          "items": {
+            "$ref": "#/definitions/CustomBotCategory"
+          }
+        }
+      }
+    },
+    "CustomBotCategory": {
+      "description": "Defines a collection of bot identifier with an associated action.",
+      "required": [
+        "action",
+        "matchConditions",
+        "priority"
+      ],
+      "properties": {
+        "name": {
```

Comment:
Should this be required? We need an identifier for logging, debugging..

Reply:
i just saw it wasnt required for custom rules so i didnt think it was necessary but i see your point i agree. not sure how the custom rules work for logging. ill make it required to be safe